### PR TITLE
Refactor native cpu adapter to new logger

### PR DESCRIPTION
**Pull request by lplewa**

### CODE REVIEW

**Comment by lplewa** [comment]
I created an octopus merge b45470f9e1a5c4bacbc210d68e900858b82fae25 of all logs changes (#1359; #1347; #1345; #1342; #1032) and tested integration with [intel/llvm repository](https://github.com/intel/llvm/pull/12580/files)

File: source/adapters/native_cpu/common.hpp [hunk 1]
```diff
@@ -18,29 +19,24 @@ extern thread_local ur_result_t ErrorMessageCode;
...
 
 #define CONTINUE_NO_IMPLEMENTATION                                             \
-  if (PrintTrace) {                                                            \
-    std::cerr << "Warning : Not Implemented : " << __FUNCTION__                \
-              << " - File : " << __FILE__;                                     \
-    std::cerr << " / Line : " << __LINE__ << std::endl;                        \
-  }                                                                            \
-  return UR_RESULT_SUCCESS;
+  do {                                                                         \
+    logger::warning("Not Implemented : {} - File : {} / Line : ",              \
```
Comment (PietroGhg) [inline source/adapters/native_cpu/common.hpp:31]: Same as above, maybe there's a `{}` missing

**Review by PietroGhg — COMMENTED** [review]
This looks good thank you, only one question: I'm trying to run a simple test built for Native CPU with `UR_LOG_LOADER="level:debug;flush:debug;output:stderr"` set, what I  expect is to see some warnings about functions that are not yet implemented in this adapter, e.g. `urPlatformGetBackendOption`, but I only get `<LOADER>[DEBUG]: Logger loader initialized successfully!`, I'm not familiar with the new logger, is there something I'm missing in the env variable?

**Comment by pbalcer** [comment]
> Native CPU with `UR_LOG_LOADER="level:debug;flush:debug;output:stderr"` set

Each component of UR has its own env variable to control verbosity of logging. This is so that it's possible for the user to see traces for the adapter, without getting spammed with potentially irrelevant messages about the loader.
In this case, you should set `UR_LOG_NATIVE_CPU` env variable.

@lplewa please add docs here https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/INTRO.rst#environment-variables (for each adapter you are changing)

**Comment by lplewa** [comment]
> @lplewa please add docs here https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/INTRO.rst#environment-variables (for each adapter you are changing)

https://github.com/oneapi-src/unified-runtime/pull/1376

**Comment by lplewa** [comment]
@PietroGhg - please review - if you are ok with changes please label it as ready to merge

**Review by PietroGhg — APPROVED** [review]
LGTM, thank you

**Pull request merged**